projects/plugins/jetpack/require-lib.php

```diff
@@ -20,16 +16,29 @@ function jetpack_require_lib( $slug ) {
 	 * @param string $lib_dir Path to the library directory.
 	 */
 	$lib_dir = apply_filters( 'jetpack_require_lib_dir', $lib_dir );
-	$choices = array(
-		"$lib_dir/$slug.php",
-		"$lib_dir/$slug/0-load.php",
-		"$lib_dir/$slug/$basename.php",
-	);
-	foreach( $choices as $file_name ) {
-		if ( is_readable( $file_name ) ) {
-			require_once $file_name;
-			return;
-		}
+
+	$loaded_key = "{$lib_dir}{$slug}";
+	if ( ! empty( $loaded[ $loaded_key ] ) ) {
+		return;
+	}
+
+	$loaded[ $loaded_key ] = true;
+
+	$file_name = "$lib_dir/$slug.php";
+	if ( is_readable( $file_name ) ) {
+		require_once $file_name;
+
+		return;
 	}
-	trigger_error( "Cannot find a library with slug $slug.", E_USER_ERROR );
+
+	$file_name = "$lib_dir/$slug/0-load.php";
+	if ( is_readable( $file_name ) ) {
+		require_once $file_name;
+
+		return;
+	}
+
+	$basename  = basename( $slug );
+	$file_name = "$lib_dir/$slug/$basename.php";
+	require_once $file_name;
```

Comment:
I note this may result in a misleading error. It'll complain that "$lib_dir/$slug/$basename.php" cannot be found without indicating that "$lib_dir/$slug.php" and "$lib_dir/$slug/0-load.php" are also options.